Migrate TCK code to Jose4j, clean up and improve TokenUtils tests

sberyozkin opened the pull request:
Fixes #72.

This PR does the following:
- Updates `TokenUtils` which has become leaner
- Renamed `DebugTest` to `TokenUtilsGenerateTokenTest`, cleaned it up, added the checks the original token claims have made it into the verified form, added an invalid 'alg' test.
- Renamed `TokenUtilsTest` to `TokenUtilsMiscMethodsTest`, removed 2 of them since there were just printing something, with `TokenUtilsGenerateTokenTest` covering well the parsing/key loading, and updated 3 remaining tests to check the key length, and added the one to check a soon to be deprecated `1024` bit length can be created as well.

As I've mentioned at #72, even though it is the library we use in our own projects, it is really a good, compact and trusted library, so IMHO it will do well for the TCK

rdebusscher commented on tck/src/test/java/org/eclipse/microprofile/jwt/tck/util/TokenUtilsGenerateTokenTest.java:
```diff
@@ -0,0 +1,152 @@
+ * Internal debugging tests
+ */
+public class TokenUtilsGenerateTokenTest {
+
```
Within `DebugTest`, there was a `testHS256` test which doesn't seems to be retained. What is the reason for this?

sberyozkin replied:
@rdebusscher yes, I was looking at it, it was just testing that Nimbus could use `HS256` without even using `TokenUtils`. And we won't support the symmetric signatures for 1.2. So I dropped it but as mentioned I've introduced an invalid `alg` test which actually stresses the relevant TokenUtils code

rdebusscher commented on tck/src/test/java/org/eclipse/microprofile/jwt/tck/util/TokenUtilsGenerateTokenTest.java:
```diff
@@ -0,0 +1,152 @@
+
+    @Test(groups = TCKConstants.TEST_GROUP_UTILS, expectedExceptions = {InvalidJwtException.class},
+            description = "Illustrate validation of iss")
+    public void testFailAlgorithn() throws Exception {
```
typo testFailAlgorith**n**

rdebusscher commented on tck/src/test/java/org/eclipse/microprofile/jwt/tck/util/TokenUtilsGenerateTokenTest.java:
```diff
@@ -0,0 +1,152 @@
+        // Confirm all the claims available in /Token1.json have made it into the verified claimSet
+
+        Assert.assertEquals(claimsSet.getClaimsMap().size(), 18);
+        Assert.assertEquals("https://server.example.com", claimsSet.getIssuer());
```
The signature is `Assert.assertEquals(Actual, Expected)` and thus parameters are swapped (also for the following lines)

rdebusscher commented on tck/src/main/java/org/eclipse/microprofile/jwt/tck/util/TokenUtils.java:
```diff
@@ -120,16 +112,19 @@ public static String generateTokenString(PrivateKey pk, String kid, String jsonR
+        if (contentIS == null) {
+            throw new IllegalStateException("Failed to find resource: " + jsonResName);
+        }
         byte[] tmp = new byte[4096];
```
This is a much easier to read way of getting the content of the InputStream in a String

> Scanner s = new Scanner(inputStream).useDelimiter("\\A");
> String result = s.hasNext() ? s.next() : "";

rdebusscher commented on tck/src/main/java/org/eclipse/microprofile/jwt/tck/util/TokenUtils.java:
```diff
@@ -120,16 +112,19 @@ public static String generateTokenString(PrivateKey pk, String kid, String jsonR
             invalidClaims = Collections.emptySet();
         }
         InputStream contentIS = TokenUtils.class.getResourceAsStream(jsonResName);
```
InputStream is never closed.

rdebusscher commented on tck/src/test/java/org/eclipse/microprofile/jwt/tck/util/TokenUtilsMiscMethodsTest.java:
```diff
@@ -0,0 +1,65 @@
+public class TokenUtilsMiscMethodsTest {
+
+    @Test(groups = TEST_GROUP_UTILS, description = "Used to generate a 2048 bit length key testing pair")
+    public void testKeyPairGeneration2048Length() throws Exception {
```
Not very useful as this just test the JDK generation of an RSA key.

sberyozkin replied:
Yeah, I was tempted to remove altogether but well, it is testing that `TokenUtils` is producing it correctly

sberyozkin replied:
The other thing this test is testing is that `2048` is enforced, same for the other test

rdebusscher commented on tck/src/test/java/org/eclipse/microprofile/jwt/tck/util/TokenUtilsMiscMethodsTest.java:
```diff
@@ -0,0 +1,65 @@
+    }
+
+    @Test(groups = TEST_GROUP_UTILS, description = "Used to generate a 1024 bit length key testing pair")
+    public void testKeyPairGeneration1024Length() throws Exception {
```
Not very useful as this just test the JDK generation of an RSA key.

sberyozkin commented:
@rdebusscher thanks, re the years, which years should we do, should we have a separate issue to change them across all the source in time for 1.2 ?

rdebusscher commented:
@sberyozkin  Regarding the years

Every new file which is added needs to have the year itself, thus 2020 now.

For every file which is updated, we need to update the license year to also include 2020 (update end year of the already present fork or add 2020 as end year).

No mass update of all files, only changed files require and update (and thus should be done in each PR)

sberyozkin commented:
@rdebusscher Thanks for the info...

sberyozkin commented:
Hi @rdebusscher I think I've addressed the comments, have a look please. I've set the year to 2020 in all the files, kept the lower margin year as these are not really new files, just renamed the test files.
Resolved few comments; those RSA key tests, yeah, they just check the utility methods do what they are expected to do. The main testing is in the generate token test method...

sberyozkin commented:
@rdebusscher Thanks
